core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -681,13 +686,20 @@ public PlanWithProperties visitFilter(FilterNode node, PreferredProperties prefe
         @Override
         public PlanWithProperties visitTableScan(TableScanNode node, PreferredProperties preferredProperties)
         {
-            if (!node.isUseConnectorNodePartitioning()) {
+            if (!isUseTableScanNodePartitioning(session)) {
                 return new PlanWithProperties(node, deriveProperties(node, ImmutableList.of()));
             }
+
             // attempt tp push preferred partitioning into the table scan
             TableScanNode partitionedNode = preferredProperties.getGlobalProperties()
                     .flatMap(Global::getPartitioningProperties)
                     .flatMap(partitioningProperties -> pushPartitioningIntoTableScan(plannerContext.getMetadata(), session, node, partitioningProperties))
+                    .map(tableScanNode -> DetermineTableScanNodePartitioning.setUseConnectorNodePartitioning(
```

Comment:
You don't need to set `withUseConnectorNodePartitioning` in `pushPartitioningIntoTableScan` since you run `setUseConnectorNodePartitioning` here that is also overriding `withUseConnectorNodePartitioning`.

You should probably return original `node` (without partitioning pushed down) if `setUseConnectorNodePartitioning` decides not to use connector partitioning

Reply:
This was a WIP and not meant to be reviewed.  The final version calls into directly into DetermineTableScanNodePartitioning to set this.